services/scan/src/store.ts

```diff
@@ -375,6 +381,95 @@ export class Store {
     );
   }
 
+  setBatchesAsBackedUp() {
+    this.setBackupTimestamp(BackupKey.Batches);
+  }
+
+  setCvrsAsBackedUp() {
+    this.setBackupTimestamp(BackupKey.Cvrs);
+  }
+
+  /**
+   * Sets the timestamp of a backup
+   */
+  private setBackupTimestamp(key: BackupKey): void {
+    this.client.run(
+      'insert or replace into backups (key, value) values (?, current_timestamp)',
+      key
+    );
+  }
+
+  /**
+   * Gets a config value by key.
+   */
+  private getBackupTimestamp(key: BackupKey): Iso8601Timestamp | undefined {
+    const row = this.client.one<[Iso8601Timestamp]>(
+      'select value from backups where key = ?',
+      key
+    ) as { value: Iso8601Timestamp } | undefined;
+    return row?.value;
+  }
+
+  /**
+   * Returns whether the appropriate backups have been completed and it is safe
+   * to unconfigure a machine/zero out data. Always returns true in test mode.
+   */
+  getCanUnconfigure(): boolean {
```

Comment:
I think this is the most important piece for reviewers to spend time on.

Specifically, if someone better at SQL is interested in helping me capture this heuristic in a single query, that would be awesome!

Reply:
there's a way to do this in two simple SQL queries, I think, which, depending on your pov, might be ugly, or might be a nice encapsulation of the whole thing.

`select max(max(created_at, deleted_at)) <= ? from sheets`

`select max(max(created_at, deleted_at)) <= ? from batches`

If both are true, you're good to go.